R/row_to_names.R

```diff
@@ -44,7 +42,16 @@ row_to_names <- function(dat, row_number, ..., remove_row = TRUE, remove_rows_ab
   } else {
     stop("row_number must be a numeric value or 'find_header'")
   }
-  new_names <- as.character(unlist(dat[row_number, ], use.names = FALSE))
+  new_names <- dat[row_number, ] %>% 
+    lapply(reduce_na) %>% 
+    as.data.frame() %>% 
+    replace(is.na(.), "") %>% 
+    lapply(paste_skip_na, collapse = sep) %>% 
+    lapply(stringr::str_replace_all, pattern = "__+", replacement = "_") %>% 
+    lapply(stringr::str_replace_all, pattern = "_$", replacement = "") %>% 
+    unlist(use.names = FALSE) %>% 
+    as.character()
```

Comment:
I think that all of this could be simplified to `sapply(dat[row_number, ], paste_skip_na, collapse = sep)`, right?  (Maybe I'm missing something.)  The downside is that when all values are `NA`, it would have a column name that is actually `NA` rather than the character string `"NA"`.  The other downside is that you'll have sequential underscores and underscores at the end in some scenarios, but I think that is the responsibility of the user to convert blanks to `NA` if they don't want them.  (Maybe that's too much effort to push to the user.)

Also, the use of `"_"` here relies on the fact that `sep = "_"`, but if I set `sep` to something else, it would not have the same type of behavior.

Reply:
I agree, I changed the code to your suggestion.
To handle NA variable names i Used `stringr::str_replace_na()`, this is a classic use case for this.